File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/math_random_int.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeMathRandomInt(statement: JSONObject, onError: (e: Exception) -> Unit): Int? {
    try {
        Log.d("hello", "execute_math_random_int")
        val pointer = statement.getJSONObject("pointer")
        val minValue = executeStatement(pointer["min"].toString()) ?: return null
        val maxValue = executeStatement(pointer["max"].toString()) ?: return null

        return (minValue.toString().toInt()..maxValue.toString().toInt()).random()
    } catch (e: Exception) {
        onError(e)
        return null
    }
}

File: piper/extensions/chrome/src/engine/statements/math_random_int.js
import { stringify } from "../util"


function mathRandomInt(a, b) {
    if (a > b) {
      var c = a;
      a = b;
      b = c;
    }
    return Math.floor(Math.random() * (b - a + 1) + a);
}

export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let minValue = await this.executeStatement(stringify(pointer.min))
        let maxValue = await this.executeStatement(stringify(pointer.max))

        return mathRandomInt(minValue, maxValue)
    } catch (e) {
        onError(e)
    }
}
